### ISSUE TITLE
simple EDM producer v0

### ISSUE THREAD

**Issue by hq9000** [hq9000/py_headless_daw#23]
now that the DAW part is roughly ready, it's time to move to introduce a higher level entity - the "producer".
As seen now, normally, producers will be written in the apps using this package, but this "Simple EDM" one will be created right here, for simplicity and demonstration purposes. Also, it's likely to require quite a bit of changes in the headless daw part itself. 

Moreover, this simple EDM producer is expected to make use of each and every relevant PHD feature, so it's a good idea to have it tested in the PHD as well. 

Long story short,  all of this above justifies the decision of NOT starting a new repo for this purpose.

## simple EDM producer

- conforms to producer interface (to be defined)
- generates randomized projects of the following kind:
  - 4 tracks: bass drum, synth1, synth2, synth3
  - 4, 16 or 32 bars length "themes"

to explore:

- what patches does amsynth have?
 
notes:

- bass drum track should use the recently added "drum synth" and use some reasonably sounding preset with some random variations.
  

upd:

what about #10 ?

**Comment by hq9000** [comment]
- [ ] drum synth plugin -> generator:  make number of oscillators configurable with parameter (enum: 1,2,3,4)

**Comment by hq9000** [comment]
[ ] see what's currently rendered with pyplot

**Comment by hq9000** [comment]
![image](https://user-images.githubusercontent.com/21345604/100302004-6d527b80-2faa-11eb-969e-216b195da62c.png)

**Comment by hq9000** [comment]
- [ ] finish refactoring regarding clip relative time

**Comment by hq9000** [comment]
patches for amsynth http://linuxsynths.com/amSynthdemos/amsynth.html

**Comment by hq9000** [comment]
compressor https://www.audiopluginsforfree.com/roughrider3/

**Comment by hq9000** [comment]
![image](https://user-images.githubusercontent.com/21345604/100691741-65278100-339a-11eb-8a4c-efad7c1930b0.png)

**Comment by hq9000** [comment]
![image](https://user-images.githubusercontent.com/21345604/100824170-b56c1500-3466-11eb-9848-7efb7e0e9ae3.png)

**Comment by hq9000** [comment]
![image](https://user-images.githubusercontent.com/21345604/100846973-368bd200-3490-11eb-9877-ec41ea7ffba9.png)

**Comment by hq9000** [comment]
notes on running/debuggin amsyth problem:

- had to install dbg into dev container
- to make breakpoints hit, had to make debug configuration like this
![image](https://user-images.githubusercontent.com/21345604/101329283-3d488980-3882-11eb-8961-7da9676644e9.png)
- the actual sound generation is in `VoiceBoard.cpp`
